apps/web/src/services/asset-inventory/collector/shared/collector-data-modal/CollectorDataModal.vue

```diff
@@ -130,10 +137,14 @@ const handleClickConfirm = async () => {
 
 /* API */
 const apiQueryHelper = new ApiQueryHelper();
-const fetchSecrets = async (provider: string) => {
-    try {
-        apiQueryHelper.setFilters([{ k: 'provider', v: provider, o: '=' }]);
+const fetchSecrets = async (provider: string, serviceAccounts?: string[]) => {
+    apiQueryHelper.setFilters([{ k: 'provider', v: provider, o: '=' }]);
 
+    if (serviceAccounts?.length) {
+        if (state.isExcludeFilter) apiQueryHelper.addFilter({ k: 'service_account_id', v: serviceAccounts, o: '!=' });
+        else apiQueryHelper.addFilter({ k: 'service_account_id', v: serviceAccounts, o: '=' });
+    }
```

Comment:
issue: It's pretty risky that there is no page limit or additional filters that can narrow down the scope in query. It causes unnecessary retrieve of data on database layer.

So I recommend to add count only option since it seems that there is no requirement for detail information of secret list.

Reply:
plz check it again!